File: Week_03/id_98/LeetCode_703_99.java
class KthLargest {


	    final PriorityQueue<Integer> pq;
	    final int k;
	    
	    public KthLargest(int k, int[] nums) {
	        this.k = k;
	        pq = new PriorityQueue<>(k);
	        for(int i : nums){
	            add(i);
	        }
	    }
	    
	    public int add(int val) {
	        if(pq.size() < k)
	            pq.offer(val);
	        else{
	            if(pq.peek() < val){
	                pq.poll();
	                pq.offer(val);
	            }
	        }
	        return pq.peek();
	    }

}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: Week_03/id_99/LeetCode_703_99.cpp
#include <queue>
using namespace std;

class KthLargest {
public:
    int topK;
    priority_queue<int,vector<int>,greater<int>> bigHeapQ;
public:
    KthLargest(int k, vector<int>& nums) {
        topK = k;
        for(int i=0;i<nums.size();++i){
            add(nums[i]);
        }
    }
    
    int add(int val) {
        if(bigHeapQ.size() < topK) {
            bigHeapQ.push(val);
        } else if(bigHeapQ.top() < val) {
            bigHeapQ.pop();
            bigHeapQ.push(val);
        }
        return bigHeapQ.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */